packages/web3-eth-accounts/src/index.js

```diff
@@ -154,6 +154,13 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
         return Promise.reject(error);
     }
 
+    if (!isNot(tx.common) && !isNot(tx.common.customChain.chainId) && !isNot(tx.chainId) && tx.chainId !== tx.common.customChain.chainId) {
```

Comment:
There is no check I could find in the flow which enforces that if `tx.common` exists then `tx.common.customChain` must exists. So in case `tx.common` exists but `tx.common.customCain` does not code will throw a runtime error. So we must add an additional check here. 

Also using so many negative assertions reduce readability. So create a small utility function `isExists` and then use it here to avoid those double negations. 

```javascript
if(isExists(tx.common) && isExists(tx.common.customChain) && isExists(tx.common.customChain.chainId) && isExists(tx.chainId) && tx.chainId !== tx.common.customChain.chainId)) {
```

Reply:
There are no double negation operators in this function and its call, 
I even personally dnt like naming of `isNot` and there are 100s of places in 1.x for **readability** improvement, that why we will only focus our energies on only functional changes in 1.x and will not overhaul it.

Reply:
I didn't said there is double negation operator 😄 

I said dual negative assertions, once from `isNot` and then `!` operator makes it difficult to read. And such few minutes of refactoring which increase readability is not an effort we should avoid for the code base we touch upon. Earlier such practice was avoided end-up in poor readable code.